src/Idris/CommandLine.idr

```diff
@@ -260,7 +260,7 @@ processArgs flag (Optional a :: as) [] f =
 processArgs flag (Required a :: as) (x :: xs) f =
   processArgs flag as xs (f x)
 processArgs flag (Optional a :: as) (x :: xs) f =
-  processArgs flag as xs (f (Just x))
+  processArgs flag as xs (f $ toMaybe (not (any (\p => isPrefixOf p x) ["-", "--"])) x)
```

Comment:
If `--` is a prefix then so is `-`. So it's enough to check for `-` only.

Reply:
Makes sense.  Thanks, I will fix it.